doc/1/controllers/document/search/index.md

```diff
@@ -0,0 +1,68 @@
+---
+code: true
+type: page
+title: SearchAsync
+description: Searches documents.
+---
+
+# SearchAsync
+
+Searches documents.
+
+There is a limit to how many documents can be returned by a single search query.
+That limit is by default setted at 10000 documents, and you can't get over it even with the from and size pagination options.
+
+:::info
+When processing a large number of documents (i.e. more than 1000), it is advised to paginate the results using [SearchResult.Next](/sdk/csharp/1/core-classes/search-result/next/) rather than increasing the size parameter.
```

Comment:
```suggestion
When processing a large number of documents (i.e. more than 1000), it is advised to paginate the results using [SearchResult.Next](/sdk/csharp/1/core-classes/search-result/next) rather than increasing the size parameter.
```